### ISSUE TITLE
[1.8.5] Chalice liquid interaction with insufficient starlight

### ISSUE THREAD

**Issue by kerzenhans** [HellFirePvP/AstralSorcery#567]
Finally had time to test the Chalice change in 1.8.5 ( see https://github.com/HellFirePvP/AstralSorcery/issues/440 )
Setup: XNet pumping 1mb of starlight/lava in 2 chalices every 200 ticks.
Result: 1mb Starlight + 1mb Lava get drained and produce a liquid interaction. 
Got 2 shale and 11 sand from my 2 minutes test.
I don't think this is intentionally, as different interactions have different volumes to consume.

**Comment by Doomgull** [comment]
It also doesnt normally ever function at a rate of 1mb of fluid into a chalice, ever, much less 1 every 200 ticks.

Does this happen whrn not using xnet im such a specific and odd fashion?

**Comment by Doomgull** [comment]
It also doesnt normally ever function at a rate of 1mb of fluid into a chalice, ever, much less 1 every 200 ticks.

Does this happen whrn not using xnet im such a specific and odd fashion?

**Comment by kerzenhans** [comment]
When I found the old issue I had a setup with a Neromantic Prime and the chalice with lava was filled by ExU2 nodes from a Mekanism tank. When I decided I had enough sand I broke the eveshifting fountains support structure, which resulted in an odd value of starlight in the chalice.
Which I think comes form the fountain extracting random values out of the chunk reservoir.
The infinity form #440 is fixed, as the chalices now get completely drained of their excess fluid although producing an interaction.

Now to test the fix I explicitly set up XNet to replicate the problematic values in the tanks. 
Which I am sure can also be achieved with every other automation/control mod capable of fluid routing in explicit value. 

If I read your code right:
chalice interaction samples an interaction for the general liquid types, 
then drain the tanks for maxDrain = interaction given drain 
(ignoring the possibility that there might be less amount of fluid in the tanks as maxDrain, which is the implementation of IFluidhandler.doDrain(int max, bool simulate) ) 
then spawn the lightshow which will trigger the drop when the entities meet.

This should also results in the possibility to get a rock crystal drop with only 10mb starlight remaining which should only be sufficient for sand.
